test/pages.js

```diff
@@ -2379,4 +2381,127 @@ describe('Pages', function() {
+  // given the rule against a page with an unpublished parent becoming published,
+  // which is a very good rule!
+  //
+  // However the doc template library does create autopublished pages as children of
+  // the home page, which is always published, so we need to handle
+  // them successfully at a low level. -Tom
+
+  describe('autopublish', function() {
+    before(async () => {
+      await t.destroy(apos);
+      apos = await t.create({
+        root: module,
+        modules: {
+          '@apostrophecms/express': {
+            options: {
+              apiKeys: {
+                [apiKey]: {
+                  role: 'admin'
+                }
+              }
+            }
+          },
+          '@apostrophecms/page': {
+            options: {
+              park: [],
+              types: [
+                {
+                  name: '@apostrophecms/home-page',
+                  label: 'Home'
+                },
+                {
+                  name: 'test-page',
+                  label: 'Test Page'
+                },
+                {
+                  name: 'autopublished-page',
+                  label: 'Autopublished Pages'
+                }
+              ],
+              publicApiProjection: {
+                title: 1,
+                _url: 1
+              }
+            }
+          },
+          'test-page': {
+            extend: '@apostrophecms/page-type'
+          },
+          'autopublished-page': {
+            extend: '@apostrophecms/page-type',
+            options: {
+              autopublish: true
+            }
+          }
+        }
+      });
+      await t.createAdmin(apos);
+    });
+
+    beforeEach(async function() {
+      await apos.doc.db.deleteMany({ type: { $in: [ 'test-page', 'autopublished-page' ] } });
+    });
+
+    it('should properly autopublish a page even if its page tree target is draft only', async function() {
+      const jar = await t.loginAs(apos, 'admin');
+      await apos.http.get('/', { jar });
+
+      await autopublishedPage('First Child');
+
+      const testItems = [
+        {
+          title: 'Page 1',
+          type: 'test-page',
+          slug: '/page1'
+        },
+        {
+          title: 'Page 2',
+          type: 'test-page',
+          slug: '/page2'
+        }
+      ];
+
+      for (const page of testItems) {
+        await postPage(page);
+      }
+
+      await autopublishedPage('Last Child');
+
+      async function autopublishedPage(title) {
+        const body = {
+          title,
+          type: 'autopublished-page',
+          slug: '/autopublished'
+        };
+
+        // Throws on failure, does not on success
+        await postPage(body);
+      }
+
+      async function postPage(body) {
+        await apos.http.post('/api/v1/@apostrophecms/page?aposMode=draft&aposLocale=en', {
+          jar,
+          body
+        });
+      }
+    });
+  });
 });
+
+function getIndexes(all, mode, pages) {
```

Comment:
Utility to reduce boilerplate and return an object with named properties for more readable tests.